just_audio/darwin/Classes/AudioPlayer.m

```diff
@@ -462,16 +470,32 @@ static void processTap(MTAudioProcessingTapRef tap, CMItemCount frameCount, MTAu
         if (unsignedSample > 255) unsignedSample = 255;
         else if (unsignedSample < 0) unsignedSample = 0;
         waveform[i] = (UInt8)unsignedSample;
+        realData[i] = unsignedSample;
+        imageData[i] = 0;
     }
-
+    
     // TODO: Take captureRate into account. Maybe let the main thread
     // periodically take samples, and we provide them here in a CMSimpleQueue.
     AudioPlayer *self = (__bridge AudioPlayer *)(storage->self);
     // NOTE: Apple recommends to NOT allocate any memory in the tap process function.
     // TODO: Check impact on performance and memory.
+    
+    // FFT
+    // TODO: Implement FFT using bit operation to speed up
+    fft(captureSize, realData,imageData);
```

Comment:
Sorry for the bad English,

Yes and I've already put it on the bottom of FFTHelper.c, in the comment, just in case if you need it.

There is a scaling problem. Because apple-based FFT is floating-point and we need **int8** for sake of android compatibility.  Thus, the scaling process is needed and I don't know how to do it correctly. So I decided to use integer-based FFT.
However, at the bottom of the line, I have to scale int to int8. Either way, I can't get rid of the scaling problem.

I have to choose between scaling float to int8 or int to int8. I think the floating range is too large (too many possible numbers), so I decided to use integer-based FFT. Another advantage is that usage of integer FFT is more portable. No need to initiate or destroy DSP engine. and If you worry about its performance, the integer-based FFT can be done with + and * which can transform into a bitshift operation which is really fast.

I think there may be a possibility to implement int8 FFT so we don't have to scale the result. But I fade difficulties on overflow problem so I stop making it.

I can switch to floating FFT (apple-based) if you think it provides a better result.

Cheers,